Mage.Sets/src/mage/cards/v/VinesoulSpider.java

```diff
@@ -0,0 +1,78 @@
+package mage.cards.v;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfEndStepTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.keyword.ReachAbility;
+import mage.cards.*;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+
+import java.util.UUID;
+
+/**
+ * @author karapuzz14
+ */
+public final class VinesoulSpider extends CardImpl {
+
+    public VinesoulSpider(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{G}");
+        this.subtype.add(SubType.ELEMENTAL);
+        this.subtype.add(SubType.SPIDER);
+
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(2);
+
+        this.addAbility(ReachAbility.getInstance());
+
+        // At the beginning of your end step, put a random land card from your library into your graveyard.
+        this.addAbility(new BeginningOfEndStepTriggeredAbility(
+                new mage.cards.v.VinesoulSpiderEffect(), TargetController.YOU, false
```

Comment:
don't include the `mage.cards.v`, the effect is already in the namespace (many instances to fix in this file)